### ISSUE TITLE
Argv input doesn't work

### ISSUE THREAD

**Issue by Sh3b0** [Sh3B0/cplus#7]
## Code to reproduce
```python
routine main(x: integer) : integer is
	println x;
	return 0;
end
```
```
$ ./cplus -d source.cp
$ ./a.out 5
```

## Expected output
Prints 5

## Actual output
Prints other random value, although the generated `ir.ll` seems correct.

**Comment by TonyDecvA180XN** [comment]
In clang (and also others C++ compilers) the main function is linked against the CRT definition:
```cpp
int main(int argc, char *argv[])
```
which is a number of arguments and array of strings, that are arguments themselves parsed by space.
Linking process happens consequently and involves one-by-one check of ABI compatility. In our case, we expect the integer variable in both function signatures, which is fine, but missing second array of strings, which is also fine, surprisingly.
That way, the variable `x` in the program is linked against `argc`. This can be demonstrated here:
![image](https://user-images.githubusercontent.com/24457508/158564955-8d5f2af3-566d-4c45-bb9e-149cfecc6728.png)
P. S. Note that first argument is always name of executable, `a.exe` in my case.

**Comment by Sh3B0** [comment]
Got it, my bad.
Well, we obviously need to implement the required behavior without having to introduce arrays of character pointers in the language.

I suggest to fix this by internally calling main() with the appropriate arguments then.

**Comment by TonyDecvA180XN** [comment]
Yeah, but the problem of static typing does not allow us to do that. CMD Arguments are always captured as string, so we would need to embed string to int conversion code into out program, which is something I don't think we have right now. Also, for unified conversion we would need to embed all these other functions such as str to float, str to bool into CPlus.
